### ISSUE TITLE
[3.3.5] DB/Quest: Freedom for all creatures

### ISSUE THREAD

**Issue by Kodekc** [TrinityCore/TrinityCore#17137]
http://www.wowhead.com/quest=2969/freedom-for-all-creatures
http://www.wowhead.com/npc=7997/captured-sprite-darter
Fixed: https://github.com/TrinityCore/TrinityCore/issues/4823
Save at least 6 Sprite Darters from capture (Not be counted quest credit)

Branch: 3.3.5
TC hash/commit: d9dcbbc

**Comment by tkrokli** [comment]
You have removed the **issue template** provided by TC, which is bad and not recommended.
I can not understand what the point of your issue is, since the
- **Current behaviour**: Tell us what happens

and
- **Expected behaviour**: Tell us what should happen instead

lines are missing from your issue post. Can you tell us what you intended to report here?

**Comment by lyn1337** [comment]
core_version: rev. 2ae85d8 2016-05-13 23:20:47 +0200 (3.3.5 branch) (Unix, Release)
db_version: TDB 335.61

I think saving the Sprite Darters doesn't count for Kodekc. *??* But it does for me – just tested.

**Comment by Kodekc** [comment]
Sorry, in a hurry in the morning...After escort Captured Sprite Darter, quest does not count

**Comment by lyn1337** [comment]
For me, it did. Checked again with a fresh (english) install. Had some wine + performance issues, hope the vid works.
https://youtu.be/4wI17goOf1A

I'm not aware of the correct behavior, but I think the quest timer should stop when six of them were fully rescued. Also, the quest doesn't fail if Kindal Moonweaver (7956) dies.

**Comment by msoky** [comment]
I think that this quest was removed in Cataclysm

**Comment by tkrokli** [comment]
http://www.wowhead.com/quest=2969/freedom-for-all-creatures#comments:id=1208410
`Since this quest has been removed in 4.0.3a, can the darter be obtained still or is it also removed?`

http://wow.gamepedia.com/Quest:Freedom_for_All_Creatures
[The subject of this article was removed from World of Warcraft in patch 4.0.3a.]

It should still be working in the 3.3.5 branch, though, even if it is removed from newer branches/clients. Kodekc did link his 3.3.5 rev. hash, so this is a valid issue post.

I just tested it now on TrinityCore rev. d812505fc29d 2016-05-16 18:45:44 +0200 (3.3.5 branch) (Win64, Release) and it appears that you are able to escort the Sprite Darters (http://www.wowhead.com/npc=5278/sprite-darter) out of the cage, at least if you're well enough geared and above level 40, to be able to fight off the Grimtotem clan members. I escorted the Sprite Darters to freedom (see image below):

![wowscrnshot_051616_223314](https://cloud.githubusercontent.com/assets/4903600/15301589/1a595a2c-1bb7-11e6-8feb-2626d7d3ebbf.jpg)

The issue seems to be that the Quest Giver, Kindal Moonweaver (http://www.wowhead.com/npc=7956/kindal-moonweaver) says she is going to be back at her hideout with Jer'kai Moonweaver (http://www.wowhead.com/npc=7957/jerkai-moonweaver), but she was not there when I rode back. Notice that this quest has a time limit of 6 minutes to complete both **freeing the sprite darters** and **returning to Kindal Moonweaver to turn in the quest**, so it failed for me at both my attempts. I have not tried to troubleshoot why Kindal Moonweaver does not seem to respawn at her original spot.

**Comment by lyn1337** [comment]
Kindal Moonweaver (7956) does spawn again, the issue with this is the timer not stopping after completing the task itself.

**Comment by tkrokli** [comment]
Thanks for confirming this really useful info. I was uncertain about the timer vs the quest giver behaviour.

**Comment by tkrokli** [comment]
Since NPC 7956 uses SAI, here are the script lines. See if you can find out why the quest credit fails.
```sql
SELECT * FROM `smart_scripts` WHERE `entryorguid` LIKE '7956%' ORDER BY `entryorguid`,`id`;
```
```
entryorguid  source_type      id    link  event_type  event_phase_mask  event_chance  event_flags  event_param1  event_param2  event_param3  event_param4  action_type  action_param1  action_param2  action_param3  action_param4  action_param5  action_param6  target_type  target_param1  target_param2  target_param3  target_x  target_y  target_z  target_o  comment
-----------  -----------  ------  ------  ----------  ----------------  ------------  -----------  ------------  ------------  ------------  ------------  -----------  -------------  -------------  -------------  -------------  -------------  -------------  -----------  -------------  -------------  -------------  --------  --------  --------  --------  ---------------------------------------------------------------------------------------------
       7956            0       0       1          19                 0           100            0          2969             0             0             0           29              4              0              0              0              0              0            7              0              0              0         0         0         0         0  Kindal Moonweaver - On Quest 'Freedom for All Creatures' Taken - Start Follow Invoker
       7956            0       1       2          61                 0           100            0             0             0             0             0           90              1              0              0              0              0              0            1              0              0              0         0         0         0         0  Kindal Moonweaver - On Quest 'Freedom for All Creatures' Taken - Remove Flag Standstate Kneel
       7956            0       2       3          61                 0           100            0             0             0             0             0            2            774              0              0              0              0              0            1              0              0              0         0         0         0         0  Kindal Moonweaver - On Quest 'Freedom for All Creatures' Taken - Set Faction 774
       7956            0       3       4          61                 0           100            0             0             0             0             0            1              0              0              0              0              0              0            7              0              0              0         0         0         0         0  Kindal Moonweaver - On Quest 'Freedom for All Creatures' Taken - Say Line 0
       7956            0       4       5          61                 0           100            0             0             0             0             0           19            512              0              0              0              0              0            1              0              0              0         0         0         0         0  Kindal Moonweaver - On Quest 'Freedom for All Creatures' Taken - Remove unit flag 512
       7956            0       5       6          61                 0           100            0             0             0             0             0           83              2              0              0              0              0              0            1              0              0              0         0         0         0         0  Kindal Moonweaver - On Quest 'Freedom for All Creatures' Taken - Remove npcflag 2
       7956            0       6       7          61                 0           100            0             0             0             0             0            8              1              0              0              0              0              0            1              0              0              0         0         0         0         0  Kindal Moonweaver - On Quest 'Freedom for All Creatures' Taken - React def
       7956            0       7       8          61                 0           100            0             0             0             0             0           64              1              0              0              0              0              0            7              0              0              0         0         0         0         0  Kindal Moonweaver - On Quest 'Freedom for All Creatures' Taken - Store target
       7956            0       8       0          61                 0           100            0             0             0             0             0           41         360000              0              0              0              0              0            1              0              0              0         0         0         0         0  Kindal Moonweaver - On Quest 'Freedom for All Creatures' Taken - Despawn In 360000 ms
       7956            0       9       0           4                 0           100            0             0             0             0             0            1              1              0              1              0              0              0            7              0              0              0         0         0         0         0  Kindal Moonweaver - On Aggro - Say Line 1
       7956            0      10       0          38                 0           100            1             1             1             0             0           80         795600              2              0              0              0              0            1              0              0              0         0         0         0         0  Kindal Moonweaver - On Data Set 1 1 - Action list
       7956            0      11       0           7                 0           100            0             0             0             0             0           90              1              0              0              0              0              0            1              0              0              0         0         0         0         0  Kindal Moonweaver - On Evade - Set stand state
       7956            0      12       0          77                 0           100            0             1             5             0             0           80         795601              2              0              0              0              0            1              0              0              0         0         0         0         0  Kindal Moonweaver - On counter=5 - Action ist
     795600            9       0       0           0                 0           100            0          1000          1000             0             0            1              2              0              0              0              0              0            1              0              0              0         0         0         0         0  Kindal Moonweaver - On Script - Say Line 2
     795600            9       1       0           0                 0           100            0             0             0             0             0           15           2969              0              0              0              0              0           18             20              0              0         0         0         0         0  Kindal Moonweaver - On Script - Quest Credit 'Freedom for All Creatures'
     795600            9       2       0           0                 0           100            0          5000          5000             0             0           41              0              0              0              0              0              0            1              0              0              0         0         0         0         0  Kindal Moonweaver - On Script - Despawn In 3000 ms
     795601            9       0       0           0                 0           100            0          1000          1000             0             0            1              3              0              0              0              0              0           12              1              0              0         0         0         0         0  Kindal Moonweaver - On Script - Say Line 3
     795601            9       1       0           0                 0           100            0             0             0             0             0            6           2969              0              0              0              0              0           18             20              0              0         0         0         0         0  Kindal Moonweaver - On Script - Quest fails
     795601            9       2       0           0                 0           100            0          3000          3000             0             0           41              0              0              0              0              0              0            1              0              0              0         0         0         0         0  Kindal Moonweaver - On Script - Despawn
```

**Comment by lyn1337** [comment]
#15279

**Comment by lyn1337** [comment]
```sql
-- fix small typo in SAI comment for creature "Kindal Moonweaver" (entry: 7956)
UPDATE `smart_scripts` SET `comment`='Kindal Moonweaver - On counter=5 - Action list' WHERE `entryorguid`=7956 and`source_type`=0 and`id`=12 and`link`=0;
```

**Comment by lyn1337** [comment]
#15279 [3.3.5] Core/quest: Timed quests shouldn't turn failed once completed
#15860 [3.3.5] Core/Item: Questitem not deleted after canceling the quest

**Comment by lyn1337** [comment]
- Quest giver "Kindal Moonweaver" is set to despawn (SMART_ACTION_FORCE_DESPAWN) in SAI after 360000ms (== Quest `TimeAllowed`), but instead she dies in place plus leaving a corpse.

- Quest giver "Kindal Moonweaver" needs the full respawn time (see below) to appear back at her spawn pos, after she despawns on quest completion. Instead, she should show up there right after her despawn.

- The "Bamboo Cage Key" (entry: 12301) (Quest `Startitem`) should be removed when the quest fails or Kindal dies. ← not completely sure about this, it's been some time since I did this on retail. But makes sense.

Suggestion: Remove the "Priority-Cosmetic" label
The quest can't be done as intended, as long as #15279 is not fully fixed. The quest giver needs 2 minutes to respawn (see above) because of the default `corpse.decay.normal` time being 60 seconds plus 60 seconds of the creatures `spawntimesecs` in `creature_template`.
As a consequence, there are only 4 minutes left for the player to _fully_ complete all tasks involved.
This includes at least:

1. Running to the cage after the quest was accepted

1.defeating all enemies on the way. while protecting the very weak Kindal (she will attack), which is known to be quite hard (see wowhead comments)

1. escorting the dragons + Kindal, defeat more enemies on the way

1. return to Kindal

This is quite tough and simply not enough time. (see tkroklis experience or my vid where I had to pre-clear the area)
